forage-android/src/main/java/com/joinforage/forage/android/core/ui/textwatcher/PinTextWatcher.kt

```diff
@@ -22,8 +19,9 @@ internal class PinTextWatcher(
     }
 
     override fun afterTextChanged(editable: Editable) {
-        val isValidAndComplete = editText.length() == 4
-        val isEmpty = editText.length() == 0
+        val len = editable.toString().length
+        val isValidAndComplete = len == 4
```

Comment:
As part of checking for validity, should we also verify it only contains numbers? I don't think we let people type non-numbers, but I'm not sure if it'd be possible for someone to paste in something invalid. I defer to your greater expertise on this codebase as to whether this is something worth guarding against or not.

Reply:
Good question! [The `RosettaPinElement`](https://github.com/teamforage/forage-android-sdk/blob/09c9a4824665842ac751b9007c76555228cb23f5/forage-android/src/main/java/com/joinforage/forage/android/ecom/ui/vault/rosetta/RosettaPinElement.kt#L158-L159) and [the `BTVaultWrapper`](https://github.com/teamforage/forage-android-sdk/blob/09c9a4824665842ac751b9007c76555228cb23f5/forage-android/src/main/java/com/joinforage/forage/android/ecom/ui/vault/bt/BTVaultWrapper.kt#L63) are where we specify "digits only" for the inputs. That is why this `PinTextWatcher`, only needs to be concerned about length. Technically, it would not be wrong to have it also check for digits only but it would be redundant.